pkg/server/tls_windows_test.go

```diff
@@ -0,0 +1,295 @@
+	serverSt.identities = append(serverSt.identities, newFakeIdentity(shouldFind))
+
+	findCert := func(systemStore, _ string) (certstore.Store, error) {
+		return serverSt, nil
+	}
+	identity, err := c.findCertificate(c.cfg.ServerSystemStore, c.cfg.ServerStore, c.cfg.ServerIssuerCommonNames, c.cfg.ServerTemplateID, nil, findCert)
+
+	require.NoError(t, err)
+	foundCert, err := identity.Certificate()
+	require.NoError(t, err)
+	require.True(t, foundCert == shouldFind)
+}
+
+func TestRegularExpression(t *testing.T) {
+	c := &winCertStoreHandler{
+		cfg: WindowsCertificateFilter{
+			ClientStore:        "My",
+			ClientSystemStore:  "LocalMachine",
+			ClientTemplateID:   "1.2.3",
+			ClientSubjectRegEx: "[Villa]",
+		},
+	}
+	var subjectRegEx *regexp.Regexp
+	subjectRegEx, err := regexp.Compile(c.cfg.ClientSubjectRegEx)
+	require.NoError(t, err)
+	c.subjectRegEx = subjectRegEx
+	serverSt := newFakeStore()
+	sc := makeCert(time.Now().Add(time.Duration(-5)*time.Minute), time.Now().Add(5*time.Minute), []int{1, 2, 3}, "BobVilla", "")
+	serverSt.identities = append(serverSt.identities, newFakeIdentity(sc))
+	findCert := func(systemStore, _ string) (certstore.Store, error) {
+		return serverSt, nil
+	}
+	identity, err := c.findCertificate(c.cfg.ClientSystemStore, c.cfg.ClientStore, c.cfg.ClientIssuerCommonNames, c.cfg.ClientTemplateID, c.subjectRegEx, findCert)
+
+	require.NoError(t, err)
+	foundCert, err := identity.Certificate()
+	require.NoError(t, err)
+	require.True(t, foundCert == sc)
+}
+
+func TestRegularExpression_Fail(t *testing.T) {
+	c := &winCertStoreHandler{
+		cfg: WindowsCertificateFilter{
+			ClientStore:        "My",
+			ClientSystemStore:  "LocalMachine",
+			ClientTemplateID:   "1.2.3",
+			ClientSubjectRegEx: "[Villa]",
+		},
+	}
+	var subjectRegEx *regexp.Regexp
+	subjectRegEx, err := regexp.Compile(c.cfg.ClientSubjectRegEx)
+	require.NoError(t, err)
+	c.subjectRegEx = subjectRegEx
+	serverSt := newFakeStore()
+	sc := makeCert(time.Now().Add(time.Duration(-5)*time.Minute), time.Now().Add(5*time.Minute), []int{1, 2, 3}, "BAD_EXAMPLE", "")
+	serverSt.identities = append(serverSt.identities, newFakeIdentity(sc))
+
+	findCert := func(systemStore, _ string) (certstore.Store, error) {
+		return serverSt, nil
+	}
+	identity, err := c.findCertificate(c.cfg.ClientSystemStore, c.cfg.ClientStore, c.cfg.ClientIssuerCommonNames, c.cfg.ClientTemplateID, c.subjectRegEx, findCert)
+
+	require.Error(t, err)
+	require.Nil(t, identity)
+
+}
+
+type fakeStore struct {
+	identities []fakeIdentity
+	closed     bool
+}
+
+func newFakeStore() fakeStore {
+	return fakeStore{
+		identities: make([]fakeIdentity, 0),
+		closed:     false,
+	}
+}
+
+func (f fakeStore) Identities() ([]certstore.Identity, error) {
+	ids := make([]certstore.Identity, len(f.identities))
+	for i, id := range f.identities {
+		ids[i] = id
+	}
+	return ids, nil
+}
+
+func (f fakeStore) Import(data []byte, password string) error {
+	panic("implement me")
```

Comment:
Should this be changed to "should not be called" if we're not intending to use this? Ditto for the other panics in this file